src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2809,6 +2809,19 @@ private CorInfoInitClassResult initClass(CORINFO_FIELD_STRUCT_* field, CORINFO_M
             }
         }
 
+#pragma warning disable CA1822 // Mark members as static
+        private CORINFO_METHOD_STRUCT_* getMethodFromDelegate(void* address, bool indirect)
```

Comment:
That'd be only right when `indirect` is false (always the case on NativeAOT but not CoreCLR) and I didn't want to have the signature different between the VMs.

Reply:
`CORINFO_OBJECT_HANDLE` is an existing abstraction for passing Object pointers over JIT/EE interface. It supports both frozen object references and regular object references. I do not see what prevents you from just using `CORINFO_OBJECT_HANDLE` here.

Reply:
You should be able to get the readonly field value as `CORINFO_OBJECT_HANDLE` using the existing `getStaticFieldContent` JIT/EE interface API.

Reply:
> You should be able to get the readonly field value as `CORINFO_OBJECT_HANDLE` using the existing `getStaticFieldContent` JIT/EE interface API.

AFAIR that wouldn't be legal cause the JIT does not run in COOP mode so the GC could invalidate the obtained reference.

Reply:
I would be legal. `CORINFO_OBJECT_HANDLE` is backed by a transient GC handle as necessary.

Reply:
> It would be legal. `CORINFO_OBJECT_HANDLE` is backed by a transient GC handle as necessary.

Ah, I missed that it does handles for moveable references. Do we want to switch to that then or keep this as is for perf reasons?

Reply:
We should use uniform representation for object handles throughout the JIT/EE interface.

In case the uniform representation is not performant enough, we should fix how it works so that it applies to all situations where it is used. I expect that that there is a lot more existing uses of `CORINFO_OBJECT_HANDLE` in a typical scenario than what's being introduced by this change.

Reply:
> We should use uniform representation for object handles throughout the JIT/EE interface.
> 
> In case the uniform representation is not performant enough, we should fix how it works so that it applies to all situations where it is used. I expect that that there is a lot more existing uses of `CORINFO_OBJECT_HANDLE` in a typical scenario than what's being introduced by this change.

Actually I see the issue I had here originally when I was thinking about using that: the JIT doesn't have the field handle here anymore outside of `DEBUG` builds, we'd need to make the handle available for all builds to use `getStaticFieldContent` here.

Reply:
@jkotas Could you say what you'd prefer me to do considering above?

Reply:
Done.